### DISCUSSION TITLE
ImGui initialized but not shown on window

### DISCUSSION THREAD

**Discussion by Can-Inal** [ocornut/imgui#6115] [New Users Build/Link/Run issues ONLY!]
I'm working on a Vulkan Graphics Engine and following a youtube tutorial "https://www.youtube.com/playlist?list=PL8327DO66nu9qYVKLDmdLW_84-yE4auCR". other than just following those tutorials i wanted to implement ImGui by myself. My Code compiles and runs and even imgui.ini is created but i cant see any ImGui window on my screen.

        VkDescriptorPoolCreateInfo poolInfo = {};
        poolInfo.sType = VK_STRUCTURE_TYPE_DESCRIPTOR_POOL_CREATE_INFO;
        poolInfo.poolSizeCount = static_cast<uint32_t>(poolSizes.size());
        poolInfo.pPoolSizes = poolSizes.data();
        poolInfo.maxSets = static_cast<uint32_t>(m_renderer.imageCount()) + 1;

        if (vkCreateDescriptorPool(m_device.device(), &poolInfo, nullptr, &descriptorPool) != VK_SUCCESS) {
            throw std::runtime_error("failed to create descriptor pool!");
        }

        QueueFamilyIndices Indices = m_device.findPhysicalQueueFamilies();

        // Setup Dear ImGui context
        IMGUI_CHECKVERSION();
        ImGui::CreateContext();
        ImGuiIO& io = ImGui::GetIO(); (void)io;
        io.ConfigFlags |= ImGuiConfigFlags_DockingEnable;
        io.ConfigFlags |= ImGuiConfigFlags_ViewportsEnable;
        //io.Fonts->AddFontFromFileTTF("../../Assets/Fonts/Roboto-Medium.ttf", 16.0f);

        // When viewports are enabled we tweak WindowRounding/WindowBg so platform windows can look identical to regular ones.
        ImGuiStyle& style = ImGui::GetStyle();
        if (io.ConfigFlags & ImGuiConfigFlags_ViewportsEnable)
        {
            style.WindowRounding = 0.0f;
            style.Colors[ImGuiCol_WindowBg].w = 1.0f;
        }

        io.DisplaySize = ImVec2(800, 600);
        io.DisplayFramebufferScale = ImVec2(1.0f, 1.0f);

        // Setup Platform/Renderer bindings
        ImGui_ImplGlfw_InitForVulkan(m_window.getWindowHandle(), true);
        ImGui_ImplVulkan_InitInfo init_info = {};
        init_info.Instance = m_device.getinstance();
        init_info.PhysicalDevice = m_device.getPhysicalDevice();
        init_info.Device = m_device.device();
        init_info.QueueFamily = Indices.graphicsFamily;
        init_info.Queue = m_device.presentQueue();
        init_info.PipelineCache = VK_NULL_HANDLE;
        init_info.DescriptorPool = descriptorPool;
        init_info.Allocator = NULL;
        init_info.MinImageCount = 2;
        init_info.ImageCount = static_cast<uint32_t>(m_renderer.imageCount());
        init_info.CheckVkResultFn = NULL;
        ImGui_ImplVulkan_Init(&init_info, m_renderer.getSwapChainRenderPass());

        // Setup Dear ImGui style
        ImGui::StyleColorsDark();

        VkCommandBuffer commandBuffer = m_device.beginSingleTimeCommands();
        ImGui_ImplVulkan_CreateFontsTexture(commandBuffer);
        m_device.endSingleTimeCommands(commandBuffer);
        ImGui_ImplVulkan_DestroyFontUploadObjects();

this is my InitImGui function. Im quite new with computer graphics and vulkan. I appreciate your helps thank you!

**Comment by emoon** [comment]
My suggestion would be that you take a look at an existing Vulkan implementation such as this https://github.com/ocornut/imgui/blob/master/examples/example_glfw_vulkan/main.cpp 

Get the example up and running and then you can compare what the example is doing and what your code may be doing differently.

**Reply by Can-Inal** [reply]
sorry for late reply, I checked this example implementation and figured out what i was missing! a silly mistake but i forgot to call `ImGui_ImplVulkan_RenderDrawData(ImGui::GetDrawData(), commandBuffer);` at the end of frame. Thanks for help :smile:
